### ISSUE TITLE
Asynchronous requests with Workers (token header changes globally)

### ISSUE THREAD

**Issue by thecritic** [Shopify/shopify_api#270]
Is it somehow possible to make requests asynchronously without the "singleton" session ? 
Scenario: I have Sidekiq worker Wa that makes a request to the ShopifyAPI with token A, another (asynchronous) worker Wb with token B, ... another worker Wx with token X. The problem is, when I set the session on one worker, all the other workers are affected by the change, i.e. all sessions operate with the same token (activate_session changes the header globally). 

Worker 1: |===[activate session(token A)]=============[Shopify Request (token B(!!!))]=>
Worker 2:        |============[activate session(token B)]========[Shopify Request (token B)]  
.....
Worker X:           [activate session(token X)]=============[Shopify Request(token B(!!!))]>

I can imagine that it's possible to do it manually, but is there any chance I can do it without going the manual route ?

**Comment by mikeyhew** [comment]
@thecritic just to be perfectly clear, you're talking about worker threads, not processes, right?

**Comment by kevinhughes27** [comment]
Why don't you want to use the temp session? 

The latest version of shopify_api should be using the latest version of activeresource which is threadsafe. Check you versions.

**Comment by thecritic** [comment]
@mikeyhew As far as I understand, we are talking both multiple processes (Sidekiq queues) and threads (per Sidekiq queue). 

@kevinhughes27 Due to a lack of documentation of this gem and a quite later answer to this issue I ended up rolling my own api wrapper. I was aware of the temp session feature at the time of my question, and I think I tried that as well but I can't remember what the issue was with it, I think there was something. My wrapper does not use ActiveResource because my domain knowledge of ActiveResource was and still is quite limited. I looked into the gem and couldn't make sense of the inner workings (again, possibly because of my lack of understanding AR).
